servers/rendering/renderer_rd/shaders/forward_mobile/scene_forward_mobile.glsl

```diff
@@ -1664,7 +1670,7 @@ void main() {
 					rim, rim_tint,
 #endif // LIGHT_RIM_USED
 #ifdef LIGHT_CLEARCOAT_USED
-					clearcoat, clearcoat_roughness, normalize(normal_interp),
+					clearcoat, clearcoat_roughness, geo_normal,
```

Comment:
Did you measure a performance difference from this? The reason normal_interp is used here is that it allows the compiler to optimize VGPR usage. Storing `geo_normal` allows us to avoid a couple calls to `normalize()` but it increases the VGPR usage during all of the hot paths in the shader which can significantly reduce performance since we already have a very high VGPR usage.

Reply:
This was actually done to fix shading on backfaces, not for performance, since `normal_interp` is the raw vertex normal while `geo_normal` is correctly inverted for backfaces. In any case I haven't found a performance difference between using `geo_normal` and using `normalize(normal_interp)`.

Reply:
Yeah, using renderdoc to dump the spriv and then loading those into AMD Gpu Analyzer we established that before and after the change 76/256 VGPRs were in use. After verifying that the shaders compared are different between this branch and master.